### PR TITLE
Merge commits for nvim 0.8 version into main.

### PR DESCRIPTION
**Pull request by ayamir**

### CODE REVIEW

**Comment by ayamir** [comment]
@Jint-lzxy @CharlesChiuGit @ClSlaid @jczhang02 Can you review it?

**Comment by CharlesChiuGit** [comment]
Sure! But I think it'll take a little while, since there's lots of commits. lol

File: lua/modules/editor/config.lua [hunk 1]
```diff
@@ -415,17 +435,18 @@ end
 function config.imselect()
 	-- fcitx5 need a manual config
 	if vim.fn.executable("fcitx5-remote") == 1 then
-		vim.cmd([[
-		let g:im_select_get_im_cmd = ["fcitx5-remote"]
-		let g:im_select_default = '1'
-		let g:ImSelectSetImCmd = {
+		vim.api.nvim_cmd({
+			[[ let g:im_select_get_im_cmd = ["fcitx5-remote"] ]],
+			[[ let g:im_select_default = '1' ]],
```
Comment (ClSlaid) [inline lua/modules/editor/config.lua:440]:
```suggestion
			[[ let g:im_select_get_im_cmd = ["fcitx5-remote"] ]],
			[[ let g:im_select_default = '1'                  ]],
```
let's make them aligned.

File: lua/core/init.lua [hunk 2]
```diff
@@ -98,17 +121,16 @@ local load_core = function()
...
 
 	require("core.options")
 	require("core.mapping")
 	require("keymap")
 	require("core.event")
 	pack.load_compile()
 
-	-- vim.cmd([[set background=light]])
-	vim.cmd([[colorscheme catppuccin]])
+	-- vim.api.nvim_command([[set background=light]])
```
Comment (ClSlaid) [inline lua/core/init.lua:132]: remove this line?

File: lua/core/init.lua [hunk 3]
```diff
@@ -23,27 +22,51 @@ local createdir = function()
...
-	vim.g.did_load_matchit = 1
-	vim.g.did_load_matchparen = 1
-	vim.g.did_load_2html_plugin = 1
-	vim.g.did_load_logiPat = 1
-	vim.g.did_load_rrhelper = 1
-	vim.g.did_load_netrw = 1
-	vim.g.did_load_netrwPlugin = 1
-	vim.g.did_load_netrwSettings = 1
-	vim.g.did_load_netrwFileHandlers = 1
+	-- Do not load menu
```
Comment (ClSlaid) [inline lua/core/init.lua:25]:
```suggestion
	-- disable menu loading
```
this should be better.

File: install/install.sh [hunk 4]
```diff
@@ -0,0 +1,182 @@
...
+  ${tty_underline}https://git-scm.com/${tty_reset}
+EOABORT
+	)"
+fi
+
+ssh -T git@github.com &>/dev/null
+if ! [ $? -eq 1 ]; then
+	abort "$(
+		cat <<EOABORT
+You must have SSH key prepared and registered at GitHub before
```
Comment (ClSlaid) [inline install/install.sh:138]:
```suggestion
You must have SSH key prepared and registered on GitHub before
```

**Review by ClSlaid — APPROVED** [review]
Nice

File: lua/core/init.lua (hunk 2 above)
Comment (CharlesChiuGit) [reply lua/core/init.lua:132]: Agreed

File: lua/core/init.lua (hunk 3 above)
Comment (CharlesChiuGit) [reply lua/core/init.lua:25]: Agreed

**Comment by jczhang02** [comment]
I have started to use the 0.8 branch config. The review operation will be performed after a period of experience. Hope all things well. 
Moreover, have anyone tested that whether `lspsaga winbar` will cause the problem mentioned in the  [https://github.com/ayamir/nvimdots/issues/242#issuecomment-1268445852](comment) .

**Comment by CharlesChiuGit** [comment]
@jczhang02 I also use lspsaga in winbar, I think it's pretty decent. However, since lspsaga is being called when `LspAttach`, the winbar will popup after a very short period of time. Personally I don't think it's not a big problem since I want a faster startup.

image

**Review by CharlesChiuGit — APPROVED** [review]
A legit merge!
Also, I think it would nice to update the readme, like screenshots and startup time etc.

I personally got ~25ms at startup time with 97 plugins, even if it's in my wsl.

image

image

**Comment by jczhang02** [comment]
> @jczhang02 I also use lspsaga in winbar, I think it's pretty decent. However, since lspsaga is being called when `LspAttach`, the winbar will popup after a very short period of time. Personally I don't think it's not a big deal since I want a faster startup. Also, I do not notice any stuttering after the winbar is loaded.
> 
> image

Good Job!
Can u share your `nvimdots` for learning how to configure `winbar` as beautiful as yours?
Thanks!

**Comment by CharlesChiuGit** [comment]
> Good Job! Can u share your `nvimdots` for learning how to configure `winbar` as beautiful as yours? Thanks!

I'll do a PR after this merge, since I don't want to mass up this merge. lmao
The PR will Includes aligning all icons thought out the whole nvim.

**Comment by Jint-lzxy** [comment]
I've been a tad bit busy here, sorry for the late reply.

LGTM 👍 Just wanna mention several minor adjustments before/after merging:
- [ ] The structure of `README` might need to be adjusted, and more importantly, we need to add the **new** installation method under `*nix`.
```sh
/bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/ayamir/nvimdots/HEAD/install/install.sh)"
```
- [ ] **?** Add an easy-to-use installation method on Windows.  I'm not good at writing batch scripts, so probably need some help on this.
- [ ] Support `git-https` cloning method in the installation script and the code, planning to finish this recently :)
- [ ] Currently, the **ONLY** location where this set of config would work is `${HOME}/.config/nvim`, but instead, we should let the user make their decision. This requires adjusting rules for directory searching of several plugins.

**Comment by jczhang02** [comment]
I think that `The core part of nvim8.0 can be merged as fast as possible`. `README`, `script` and etc. can be modified later.

**Review by Jint-lzxy — APPROVED** [review]
Did a full test on `MacOS`, everything _(including keybindings, old functionalities, new features, etc.)_ worked as expected  🎉

**Pull request merged**